<launch>
    <!-- Uncompress images for color and depth -->
    <node name="republish_color" type="republish" pkg="image_transport" args="compressed in:=/d435i/color/image_raw raw out:=/d435i/color/image_raw" />
    <node name="republish_depth" type="republish" pkg="image_transport" args="compressed in:=/d435i/aligned_depth_to_color/image_raw raw out:=/d435i/aligned_depth_to_color/image_raw"/>
    <!-- Uncompress images for stereo_image_rect -->
    <node name="republish_left" type="republish" pkg="image_transport" args="compressed in:=/d435i/infra1/image_rect_raw raw out:=/d435i/infra1/image_rect_raw" />
    <node name="republish_right" type="republish" pkg="image_transport" args="compressed in:=/d435i/infra2/image_rect_raw raw out:=/d435i/infra2/image_rect_raw" />

    <!-- stereo image proc --> 
    <group name="stereo">
        <node name="stereo_image_proc" type="stereo_image_proc" pkg="stereo_image_proc" output="screen" >
            <remap from="left/image_raw"    to="/d435i/infra1/image_rect_raw"/>
            <remap from="left/camera_info"  to="/d435i/infra1/camera_info"/>
            <remap from="right/image_raw"   to="/d435i/infra2/image_rect_raw"/>
            <remap from="right/camera_info" to="/d435i/infra2/camera_info"/>
            <remap from="/points2"          to="/stereo/pointcloud"/>
        </node>

        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find neptune_ros)/rviz/pointcloud.rviz" />
    </group>

</launch>
